File: Latihan_B/python/Pc.py
from Processor import Processor
from Disk import Disk
from Ram import Ram

class Pc:
    # atribut private
    __procie = Processor()
    __storage = Disk()
    __memory = Ram()
    __totalPrice = 0

    # constructor
    def __init__(self, procie, storage, memory, totalPrice=0):
        self.__procie = procie
        self.__storage = storage
        self.__memory = memory
        self.__totalPrice = totalPrice

    # method set untuk procie, storage, memory, dan totalPrice
    def setProcie(self, procie):
        self.__procie = procie
    def setStorage(self, storage):
        self.__storage = storage
    def setMemory(self, memory):
        self.__memory = memory
    def setPrice(self, totalPrice):
        self.__totalPrice = totalPrice

    # method get untuk procie, storage, memory, dan totalPrice
    def getProcie(self):
        return self.__procie
    def getStorage(self):
        return self.__storage
    def getMemory(self):
        return self.__memory
    def getTotalPrice(self):
        return self.__totalPrice
    
    # method untuk output Pc
    def printComputer(self):
        self.__procie.printProcie()
        print("============================")
        self.__storage.printDisk()
        print("============================")
        self.__memory.printRam()
        print("============================")
        print("Total Price : " + "Rp. " + str(self.getTotalPrice()))

File: Latihan_A/Pc.cpp
class Pc
{
    private:
        // atribut
        Processor procie;
        Disk storage;
        Ram memory;
        int totalPrice;

    public:
        // constructor
        Pc(){}
        Pc(Processor procie, Disk storage, Ram memory, int totalPrice) {
            this->procie = procie;
            this->storage = storage;
            this->memory = memory;
            this->totalPrice = totalPrice;
        }

        // method set untuk procie, storage, memory, dan totalPrice
        void setProcie(Processor procie) {
            this->procie = procie;
        }
        void setStorage(Disk storage) {
            this->storage = storage;
        }
        void setMemory(Ram memory) {
            this->memory = memory;
        }
        void setTotalPrice(int totalPrice) {
            this->totalPrice = totalPrice;
        }
        
        // method get untuk procie, storage, memory, dan totalPrice
        Processor getProcie() {
            return this->procie;
        }
        Disk getStorage() {
            return this->storage;
        }
        Ram getMemory() {
            return this->memory;
        }
        int getTotalPrice() {
            return this->totalPrice;
        }

        // method output untuk Pc
        void printComputer(){
            this->procie.printProcie();
            cout << "============================" << endl;
            this->storage.printDisk();
            cout << "============================" << endl;
            this->memory.printRam();
            cout << "============================" << endl;
            cout << "Total Price : " << "Rp. " << this->totalPrice << endl;
        }

        // destructor
        ~Pc(){}
};